ktor-client/ktor-client-netty/jvm/src/io/ktor/client/engine/netty/NettyClientInitializer.kt

```
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2014-2025 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.netty
+
+import io.ktor.util.date.*
+import io.netty.channel.*
+import io.netty.channel.socket.*
+import io.netty.handler.codec.http.*
+import io.netty.handler.ssl.*
+import kotlin.coroutines.*
+
+/**
+ * Channel initializer for Netty HTTP client.
+ * Configures the pipeline with SSL/TLS, HTTP codec, and response handler.
+ *
+ * @param sslCtx SSL context for secure connections, null for plain HTTP
+ * @param host Target host (used for SNI)
+ * @param port Target port
+ * @param callContext Coroutine context for the request
+ * @param requestData Request data from Ktor
+ * @param requestTime Request timestamp
+ * @param useHttp2 Whether HTTP/2 is requested (currently falls back to HTTP/1.1)
+ */
+internal class NettyClientInitializer(
+    private val sslCtx: SslContext?,
+    private val host: String,
+    private val port: Int,
+    private val callContext: CoroutineContext,
+    private val requestTime: GMTDate,
+    private val useHttp2: Boolean = false
+) : ChannelInitializer<SocketChannel>() {
+
+    override fun initChannel(ch: SocketChannel) {
+        val pipeline = ch.pipeline()
+
+        if (sslCtx != null) {
+            configureSslHandler(pipeline)
+
+            // TODO: Implement HTTP/2 support
+            // When useHttp2 is true and ALPN negotiates HTTP/2, we should configure
+            // an HTTP/2 pipeline. Currently, we always fall back to HTTP/1.1.
+            // See HTTP2_INVESTIGATION.md for details on the implementation challenge.
+            if (useHttp2) {
+                // ALPN is configured in SslContext (NettyHttpEngine.kt:117-129)
+                // Future: Add ApplicationProtocolNegotiationHandler here
+                // For now, fall back to HTTP/1.1
+                configureHttp1Pipeline(pipeline)
```

let's throw an exception when it is not implemented